pkg/ingester/ingester_v2.go

```diff
@@ -1236,6 +1246,10 @@ func (i *Ingester) v2AllUserStats(ctx context.Context, req *client.UserStatsRequ
 	i.userStatesMtx.RLock()
 	defer i.userStatesMtx.RUnlock()
 
+	if i.TSDBState.closed {
```

Comment:
If we return error here, distributor will also report error to user. Would it be better to return empty response?

Reply:
That's a really good point. I've changed it so that if the ingester is stopping, then the queries will log a message and return an empty response.